### PR TITLE
ensure item exists in stopList before attempting to get plugin data

### PR DESCRIPTION
**Pull request by joeyyang**
This PR makes the following change:
- filter out items in `stopList` that no longer exist 
  - fixes `refreshStopListOrder` bug in Specter 1.0, where if items in `stopList` were no longer in the document, `refreshStopListOrder` would fail and Specter would fail to open
joey-fix-refreshstoplistorder

### CODE REVIEW

File: specter1/src/App.ts [hunk 1]
```diff
@@ -1510,17 +1510,19 @@ export default class App {
...
-      const nodeData = JSON.parse(node.getPluginData(DATA_KEYS[`${type}NodeData`]));
-      node.setPluginData(
-        DATA_KEYS[`${type}NodeData`],
-        JSON.stringify({ ...nodeData, annotationText: position.toString() }),
-      );
-      updateAnnotationNum(item.id, position.toString(), trackingData);
-      return ({ ...item, position });
-    });
+    const reorderedList = stopList.sort(sortByPosition)
+      .filter(item => !!figma.getNodeById(item.id))
```
Comment (joeyyang) [inline specter1/src/App.ts:1514]: this could also just be `.filter(item => figma.getNodeById(item.id)`, but I wanted to make the truthiness explicit

Comment (joeyyang) [reply specter1/src/App.ts:1514]: do you think `stopList` has to be updated anywhere else? should this filter happen somewhere before this?

Comment (gblakeman) [reply specter1/src/App.ts:1514]: I agree with your reading of `refreshStopListOrder:1528`. Generally, Specter 1.0 has really good protections around recovering from stuff that no longer exists on the artboard, so beyond the test file we have access to, I’m having trouble even recreating the issue. I think we wait and see if it pops up again elsewhere before adjusting `stopList` in other places.

File: specter1/src/App.ts [hunk 2]
```diff
@@ -1510,17 +1510,20 @@ export default class App {
...
-      const nodeData = JSON.parse(node.getPluginData(DATA_KEYS[`${type}NodeData`]));
-      node.setPluginData(
-        DATA_KEYS[`${type}NodeData`],
-        JSON.stringify({ ...nodeData, annotationText: position.toString() }),
-      );
-      updateAnnotationNum(item.id, position.toString(), trackingData);
-      return ({ ...item, position });
-    });
+    const reorderedList = stopList
+      .filter(item => !!figma.getNodeById(item.id))
```
Comment (gblakeman) [inline specter1/src/App.ts:1514]: The `!!` is throwing me a bit. I’m ok with either version, but could you clarify with a comment?:

```ts
.filter(item => figma.getNodeById(item.id) // filter out nodes that we can no longer find
```

**Review by gblakeman — APPROVED** [review]
🌈🦄

**Pull request merged**